codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/ActionTestKitGenerator.scala

```diff
@@ -0,0 +1,199 @@
+          |
+          |$imports
+          |
+          |public class $testClassName {
+          |
+          |  @Test
+          |  public void exampleTest() {
+          |    CounterJournalToTopicActionTestKit testKit = CounterJournalToTopicActionTestKit.of(CounterJournalToTopicAction::new);
+          |    // use the testkit to execute a command
+          |    // ActionResult<SomeResponse> result = testKit.someOperation(SomeRequest);
+          |    // verify the response
+          |    // SomeResponse actualResponse = result.getReply();
+          |    // assertEquals(expectedResponse, actualResponse);
+          |  }
+          |
+          |  ${Format.indent(generateTestingServices(service), 2)}
+          |
+          |}
+          |""".stripMargin
+  }
+
+  def generateServices(service: ModelBuilder.ActionService): String = {
+    require(!service.commands.isEmpty, "empty `commands` not allowed")
+
+    service.commands
+      .map { command =>
+        s"""|public ActionResult<${selectOutput(command)}> ${lowerFirst(command.name)}(${selectInput(
+          command)} ${lowerFirst(command.inputType.protoName)}) {
+            |  Action.Effect<${selectOutput(command)}> effect = createAction().${lowerFirst(command.name)}(${lowerFirst(
+          command.inputType.protoName)});
+            |  return interpretEffects(effect);
+            |}
+            |""".stripMargin + "\n"
+      }
+      .mkString("")
+  }
+
+  def generateTestingServices(service: ModelBuilder.ActionService): String = {
+    service.commands
+      .map { command =>
+        s"""|@Test
+            |public void ${lowerFirst(command.name)}Test() {
+            |  ${service.className}TestKit testKit = ${service.className}TestKit.of(${service.className}::new);
+            |  // ActionResult<${selectOutput(command)}> result = testKit.${lowerFirst(command.name)}(${selectInput(
+          command)}.newBuilder()...build());
+            |}
+            |""".stripMargin + "\n"
+      }
+      .mkString("")
+  }
+
+  def selectOutput(command: ModelBuilder.Command): String = {
+    val parent = command.outputType.parent
+    s"${filterJavaOuterClassname(parent.javaOuterClassnameOption)}${command.outputType.name}"
+
+  }
+
+  def selectInput(command: ModelBuilder.Command): String = {
+    val parent = command.inputType.parent
+    s"${filterJavaOuterClassname(parent.javaOuterClassnameOption)}${command.inputType.name}"
+
+  }
+
+  def filterJavaOuterClassname(javaOuterClassnameOption: Option[String]): String = {
+    javaOuterClassnameOption match {
+      case None => ""
+      case Some(value) =>
+        if (value == "EmptyProto" || value == "AnyProto")
+          "" //TODO review this. Might not be needed if proto are 'well' defined.
+        // original .proto file on test might be badly defined
+        // e.g. `rpc Ignore(google.protobuf.Any) returns (google.protobuf.Empty)` might have to be `rpc Ignore(com.google.protobuf.Any) returns (com.google.protobuf.Empty)`
```

Comment:
This is something that I think works in the other generators, so why is it special here, or is it broken there as well?

Reply:
Is there no more. You're totally right is wasn't needed